server/src/org/apache/cloudstack/outofbandmanagement/OutOfBandManagementServiceImpl.java

```diff
@@ -0,0 +1,532 @@
+    public OutOfBandManagementResponse enableDisableOutOfBandManagement(final Host host, final boolean enabled) {
+        final OutOfBandManagement outOfBandManagementConfig = outOfBandManagementDao.findByHost(host.getId());
+        if (outOfBandManagementConfig == null) {
+            final OutOfBandManagementResponse response = new OutOfBandManagementResponse(null);
+            response.setSuccess(false);
+            response.setResultDescription("Out-of-band management is not configured for the host. Please configure the host before enabling/disabling it.");
+            return response;
+        }
+        hostAlertCache.invalidate(host.getId());
+        outOfBandManagementConfig.setEnabled(enabled);
+        boolean updateResult = outOfBandManagementDao.update(outOfBandManagementConfig.getId(), (OutOfBandManagementVO) outOfBandManagementConfig);
+        if (updateResult && !enabled) {
+            transitionPowerStateToDisabled(Collections.singletonList(host));
+        }
+        return buildEnableDisableResponse(enabled && updateResult);
+    }
+
+    @Override
+    @ActionEvent(eventType = EventTypes.EVENT_HOST_OUTOFBAND_MANAGEMENT_CONFIGURE, eventDescription = "updating out-of-band management configuration")
+    public OutOfBandManagementResponse configureOutOfBandManagement(final Host host, final ImmutableMap<OutOfBandManagement.Option, String> options) {
+        OutOfBandManagement outOfBandManagementConfig = outOfBandManagementDao.findByHost(host.getId());
+        if (outOfBandManagementConfig == null) {
+            outOfBandManagementConfig = outOfBandManagementDao.persist(new OutOfBandManagementVO(host.getId()));
+        }
+        outOfBandManagementConfig = updateConfig(outOfBandManagementConfig, options);
+        if (Strings.isNullOrEmpty(outOfBandManagementConfig.getDriver()) || !outOfBandManagementDriversMap.containsKey(outOfBandManagementConfig.getDriver().toLowerCase())) {
+            throw new CloudRuntimeException("Out-of-band management driver is not available. Please provide a valid driver name.");
+        }
+
+        boolean updatedConfig = outOfBandManagementDao.update(outOfBandManagementConfig.getId(), (OutOfBandManagementVO) outOfBandManagementConfig);
+        CallContext.current().setEventDetails("host id:" + host.getId() + " configuration:" + outOfBandManagementConfig.getAddress() + ":" + outOfBandManagementConfig.getPort());
+
+        if (!updatedConfig) {
+            throw new CloudRuntimeException("Failed to update out-of-band management config for the host in the database.");
+        }
+
+        String result = "Out-of-band management successfully configured for the host";
+        LOG.debug(result);
+
+        final OutOfBandManagementResponse response = new OutOfBandManagementResponse(outOfBandManagementDao.findByHost(host.getId()));
+        response.setResultDescription(result);
+        response.setSuccess(true);
+        return response;
```

Comment:
that method is only for enable/disable APIs. Also, this code accepts the config to populate other fields; while enable/disable response helper only cares about enable/disable operation status (true/false) and not the result description or other fields